### ISSUE TITLE
steam-deck planner can't install with `--determinate`

### ISSUE THREAD

**Issue by mbcoguno** [DeterminateSystems/nix-installer#1285]
nix-installer's steam-deck planner can't install `determinate-nixd` because `/usr/local/bin` is readonly. Is there has method to put that binary in other place? Like previous `/nix/determinate`?

P.S. not using `steamos-devmode` or `steamos-readonly`.

**Comment by grahamc** [comment]
Oh interesting ... good to know. Hm. I don't have a steam deck myself, which makes this a bit trickier

**Comment by ironicbadger** [comment]
Hi there @grahamc 👋 
Logs from an install. Some issue with `/nix` on a totally factory reset deck with everything up to date as of today.

## determinate "mode"

```
 ssh deck@10.42.7.225
deck@10.42.7.225's password:
Last login: Mon Feb 10 16:52:54 2025 from 10.42.7.29
(1)(deck@steamdeck ~)$ curl -L https://install.determinate.systems/nix | sh -s -- install steam-deck
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 17513  100 17513    0     0  70840      0 --:--:-- --:--:-- --:--:-- 70840
info: downloading installer https://install.determinate.systems/nix/tag/v0.36.1/nix-installer-x86_64-linux
 INFO nix-installer v0.36.1
`nix-installer` needs to run as `root`, attempting to escalate now via `sudo`...
[sudo] password for deck:
 INFO nix-installer v0.36.1

Cut the fuss with Determinate Nix?

Proceed? ([Y]es/[n]o/[e]xplain): n
Nix install plan (v0.36.1)
Planner: steam-deck (with default settings)

Planned actions:
* Run `systemctl daemon-reload`
* Extract the bundled Nix (originally from /nix/store/aqilnsjd2z11gzzhbcwswxhcrzjwqaf7-nix-binary-tarball-2.26.1/nix-2.26.1-x86_64-linux.tar.xz)
* Create a directory tree in `/nix`
* Synchronize /nix/var ownership
* Move the downloaded Nix into `/nix`
* Synchronize /nix/store ownership
* Create build users (UID 30001-30032) and group (GID 30000)
* Setup the default Nix profile
* Place the Nix configuration in `/etc/nix/nix.conf`
* Configure the shell profiles
* Configure upstream Nix daemon service
* Enable (and start) the systemd unit `ensure-symlinked-units-resolve.service`
* Remove directory `/nix/temp-install-dir`
* Run `systemctl daemon-reload`


Proceed? ([Y]es/[n]o/[e]xplain): y
 INFO Step: Run `systemctl daemon-reload`
 INFO Step: Clean the `/home/.steamos/offload/nix` directory
 INFO Step: Ensure SteamOS's `/nix` directory exists
 INFO Step: Enable (and start) the systemd unit `nix.mount`
 INFO Step: Create or overwrite file `/etc/atomic-update.conf.d/nix-installer.conf`
 INFO Step: Create or overwrite file `/etc/systemd/system/ensure-symlinked-units-resolve.service`
 INFO Step: Provision Nix
ERROR Error saving receipt: RecordingReceipt("/nix/receipt.tmp", Os { code: 2, kind: NotFound, message: "No such file or directory" })
ERROR
   0: Install failure
   1: Error executing action
   2: Action `provision_nix` errored
   3: Action `create_nix_tree` errored
   4: Action `create_directory` errored
   5: Creating directory `/nix/var`
   6: No such file or directory (os error 2)

Location:
   /build/nix-installer-source/src/cli/subcommand/install.rs:316

Backtrace omitted. Run with RUST_BACKTRACE=1 environment variable to display it.
Run with RUST_BACKTRACE=full to include source snippets.

Consider reporting this error using this URL: https://github.com/DeterminateSystems/nix-installer/issues/new?title=%3Cautogenerated-issue%3E&body=%23%23+Error%0A%60%60%60%0AError%3A+%0A+++0%3A+Install+failure%0A+++1%3A+Error+executing+action%0A+++2%3A+Action+%60provision_nix%60+errored%0A+++3%3A+Action+%60create_nix_tree%60+errored%0A+++4%3A+Action+%60create_directory%60+errored%0A+++5%3A+Creating+directory+%60%2Fnix%2Fvar%60%0A+++6%3A+No+such+file+or+directory+%28os+error+2%29%0A%60%60%60%0A%0A%23%23+Metadata%0A%7Ckey%7Cvalue%7C%0A%7C--%7C--%7C%0A%7C**version**%7C0.36.1%7C%0A%7C**os**%7Clinux%7C%0A%7C**arch**%7Cx86_64%7C%0A
Installation failure, offering to revert...
Nix uninstall plan (v0.36.1)

Planner: steam-deck (with default settings)

Planned actions:
* Remove the directory tree in `/nix`
* Delete file `/etc/systemd/system/ensure-symlinked-units-resolve.service`
* Delete file `/etc/atomic-update.conf.d/nix-installer.conf`
* Clean the `/home/.steamos/offload/nix` directory
* Run `systemctl daemon-reload`


Proceed? ([Y]es/[n]o/[e]xplain): y
 INFO Revert: Run `systemctl daemon-reload`
 INFO Revert: Remove directory `/nix/temp-install-dir`
 INFO Revert: Enable (and start) the systemd unit `ensure-symlinked-units-resolve.service`
 INFO Revert: Configure upstream Nix daemon service
 INFO Revert: Configure Nix
 INFO Revert: Create build users (UID 30001-30032) and group (GID 30000)
 INFO Revert: Provision Nix
 INFO Revert: Create or overwrite file `/etc/systemd/system/ensure-symlinked-units-resolve.service`
 INFO Revert: Create or overwrite file `/etc/atomic-update.conf.d/nix-installer.conf`
 INFO Revert: Enable (and start) the systemd unit `nix.mount`
 INFO Revert: Ensure SteamOS's `/nix` directory exists
 INFO Revert: Clean the `/home/.steamos/offload/nix` directory
 INFO Revert: Run `systemctl daemon-reload`
Error:
   0: Multiple errors

Location:
   /build/nix-installer-source/src/cli/subcommand/install.rs:348

Error:
   0: Action `create_directory` errored
   1: Read path `/nix/var`
   2: No such file or directory (os error 2)
```

## regular mode (also fails)

```
(1)(deck@steamdeck ~)$ curl -L https://install.determinate.systems/nix | sh -s -- install steam-deck
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 17513  100 17513    0     0  45851      0 --:--:-- --:--:-- --:--:-- 45851
info: downloading installer https://install.determinate.systems/nix/tag/v0.36.1/nix-installer-x86_64-linux
 INFO nix-installer v0.36.1
`nix-installer` needs to run as `root`, attempting to escalate now via `sudo`...
 INFO nix-installer v0.36.1

Install Determinate Nix?

Proceed? ([Y]es/[n]o/[e]xplain): n
Nix install plan (v0.36.1)
Planner: steam-deck (with default settings)

Planned actions:
* Run `systemctl daemon-reload`
* Create or overwrite file `/etc/atomic-update.conf.d/nix-installer.conf`
* Create or overwrite file `/etc/systemd/system/ensure-symlinked-units-resolve.service`
* Extract the bundled Nix (originally from /nix/store/aqilnsjd2z11gzzhbcwswxhcrzjwqaf7-nix-binary-tarball-2.26.1/nix-2.26.1-x86_64-linux.tar.xz)
* Create a directory tree in `/nix`
* Synchronize /nix/var ownership
* Move the downloaded Nix into `/nix`
* Synchronize /nix/store ownership
* Create build users (UID 30001-30032) and group (GID 30000)
* Setup the default Nix profile
* Place the Nix configuration in `/etc/nix/nix.conf`
* Configure the shell profiles
* Configure upstream Nix daemon service
* Enable (and start) the systemd unit `ensure-symlinked-units-resolve.service`
* Remove directory `/nix/temp-install-dir`
* Run `systemctl daemon-reload`


Proceed? ([Y]es/[n]o/[e]xplain): y
 INFO Step: Run `systemctl daemon-reload`
 INFO Step: Clean the `/home/.steamos/offload/nix` directory
 INFO Step: Ensure SteamOS's `/nix` directory exists
 INFO Step: Enable (and start) the systemd unit `nix.mount`
 INFO Step: Create or overwrite file `/etc/atomic-update.conf.d/nix-installer.conf`
 INFO Step: Create or overwrite file `/etc/systemd/system/ensure-symlinked-units-resolve.service`
 INFO Step: Provision Nix
ERROR Error saving receipt: RecordingReceipt("/nix/receipt.tmp", Os { code: 2, kind: NotFound, message: "No such file or directory" })
ERROR
   0: Install failure
   1: Error executing action
   2: Action `provision_nix` errored
   3: Action `create_nix_tree` errored
   4: Action `create_directory` errored
   5: Creating directory `/nix/var`
   6: No such file or directory (os error 2)

Location:
   /build/nix-installer-source/src/cli/subcommand/install.rs:316

Backtrace omitted. Run with RUST_BACKTRACE=1 environment variable to display it.
Run with RUST_BACKTRACE=full to include source snippets.

Consider reporting this error using this URL: https://github.com/DeterminateSystems/nix-installer/issues/new?title=%3Cautogenerated-issue%3E&body=%23%23+Error%0A%60%60%60%0AError%3A+%0A+++0%3A+Install+failure%0A+++1%3A+Error+executing+action%0A+++2%3A+Action+%60provision_nix%60+errored%0A+++3%3A+Action+%60create_nix_tree%60+errored%0A+++4%3A+Action+%60create_directory%60+errored%0A+++5%3A+Creating+directory+%60%2Fnix%2Fvar%60%0A+++6%3A+No+such+file+or+directory+%28os+error+2%29%0A%60%60%60%0A%0A%23%23+Metadata%0A%7Ckey%7Cvalue%7C%0A%7C--%7C--%7C%0A%7C**version**%7C0.36.1%7C%0A%7C**os**%7Clinux%7C%0A%7C**arch**%7Cx86_64%7C%0A
Installation failure, offering to revert...
Nix uninstall plan (v0.36.1)

Planner: steam-deck (with default settings)

Planned actions:
* Remove the directory tree in `/nix`
* Delete file `/etc/systemd/system/ensure-symlinked-units-resolve.service`
* Delete file `/etc/atomic-update.conf.d/nix-installer.conf`
* Clean the `/home/.steamos/offload/nix` directory
* Run `systemctl daemon-reload`


Proceed? ([Y]es/[n]o/[e]xplain): y
 INFO Revert: Run `systemctl daemon-reload`
 INFO Revert: Remove directory `/nix/temp-install-dir`
 INFO Revert: Enable (and start) the systemd unit `ensure-symlinked-units-resolve.service`
 INFO Revert: Configure upstream Nix daemon service
 INFO Revert: Configure Nix
 INFO Revert: Create build users (UID 30001-30032) and group (GID 30000)
 INFO Revert: Provision Nix
 INFO Revert: Create or overwrite file `/etc/systemd/system/ensure-symlinked-units-resolve.service`
 INFO Revert: Create or overwrite file `/etc/atomic-update.conf.d/nix-installer.conf`
 INFO Revert: Enable (and start) the systemd unit `nix.mount`
 INFO Revert: Ensure SteamOS's `/nix` directory exists
 INFO Revert: Clean the `/home/.steamos/offload/nix` directory
 INFO Revert: Run `systemctl daemon-reload`
Error:
   0: Multiple errors

Location:
   /build/nix-installer-source/src/cli/subcommand/install.rs:348

Error:
   0: Action `create_directory` errored
   1: Read path `/nix/var`
   2: No such file or directory (os error 2)
```

**Comment by ironicbadger** [comment]
Interestingly I was able to get the "normal" installer to work by unmounting `/nix` manually and rerunning.

My good friend Claude did some debugging for me, maybe it helps here.

https://shareclaude.pages.dev/c/jjts72jzy95xsq488vfmnd2q

**Comment by mbcoguno** [comment]
Right now, I recommend using NixOS's forks. `determinate-nixd` on SteamOS is painful.

## tl;dr
```sh
sudo umount /nix
curl -sSfL https://artifacts.nixos.org/nix-installer | sh -s -- install steam-deck --enable-flakes
```

P.S. `--prefer-upstream-nix` still available and can use. But it will removed at any time.
